packages/hash/api/src/graphql/resolvers/task-executor/index.ts

```diff
@@ -172,45 +149,145 @@ export const executeGithubReadTask: ResolverFn<
+      config,
+    );
+    logger.debug(
+      `Received ${airbyteRecords.length} records from Github. Traversing and rewriting into the type system.`,
+    );
+
+    const streamToKeyToPropertyTypes: Record<
+      string,
+      Record<string, PropertyType>
+    > = {};
+    const streamToEntityTypes: Record<string, EntityType> = {};
+    const entityTypeIdToEntityProperties: Record<
+      VersionedUri,
+      PropertyObject[]
+    > = {};
+
+    for (const record of airbyteRecords) {
+      const streamName = record.stream;
+
+      const streamKeyMap = streamToKeyToPropertyTypes[streamName] ?? {};
+      const entityType = streamToEntityTypes[streamName];
+      const entityProperties = record.data;
+
+      const {
+        entityProperties: updatedEntityProperties,
+        entityType: updatedEntityType,
+      } = rewriteEntityPropertiesInTypeSystem(
+        entityProperties as JsonObject,
+        streamKeyMap,
+        entityType,
+        streamName,
+        "Github",
+        user.shortname,
       );
-      logger.debug(`Received ${airbyteRecords.length} records from Github`);
-
-      const existingEntityChecker = await CachedEntityTypes(graphApi, user);
-      for (const record of airbyteRecords) {
-        const entityTypeName = streamNameToEntityTypeName(record.stream);
-        /** @todo - Check if entity already exists */
-        const entityType = existingEntityChecker.getExisting(entityTypeName);
-        if (!entityType) {
-          throw new Error(
-            `Couldn't find EntityType for ingested data with name: ${entityTypeName}`,
-          );
-        }
-
-        /** @todo - check primary key to see if entity already exists */
-        // Insert the entity
-        const entity = await createEntity(
-          { graphApi },
-          {
-            ownedById: user.accountId as OwnedById,
-            actorId: user.accountId,
-            entityType,
-            properties: record.data as PropertyObject,
-          },
-        );
 
-        createdEntities.push(entity.metadata.editionId.baseId);
+      if (entityTypeIdToEntityProperties[updatedEntityType.$id] !== undefined) {
+        entityTypeIdToEntityProperties[updatedEntityType.$id]!.push(
+          updatedEntityProperties,
+        );
+      } else {
+        entityTypeIdToEntityProperties[updatedEntityType.$id] = [
+          updatedEntityProperties,
+        ];
       }
 
-      logger.debug(`Inserted ${createdEntities.length} entities from Github`);
-      return JSON.stringify({ createdEntities });
-    } catch (err: any) {
-      throw new ApolloError(`Task-execution failed: ${err}`);
+      streamToKeyToPropertyTypes[streamName] = streamKeyMap;
+      streamToEntityTypes[streamName] = updatedEntityType;
     }
+
+    logger.debug(
+      "Finished rewriting data into entities and types, inserting into the Graph.",
+    );
+
+    const propertyTypeMap: Record<
+      VersionedUri,
+      { schema: PropertyType; created: boolean }
```

Comment:
The `created` property here and below seem to be unused?

Reply:
<img width="533" alt="image" src="https://user-images.githubusercontent.com/25749103/211013180-dabdd16f-2157-49dd-b545-38978564c4db.png">

Reply:
Unless you're referring to something else? :)